internal/api/event.go

```diff
@@ -21,7 +21,14 @@ func FormatEventControllerActor(name string) string {
 // 3. If the subject is available, it returns subject in "subject:<subject>" format.
 // 4. Otherwise, it returns EventActorUnknown.
 func FormatEventUserActor(u user.Info) string {
-	var email, subject string
+	var email, subject, username string
+	if u.Username != "" && u.Username != "email" {
+		if usernameClaim, ok := u.Claims[u.Username]; ok {
+			if usernameStr, ok := usernameClaim.(string); ok {
+				username = usernameStr
+			}
+		}
+	}
```

Comment:
I think this whole function can be boiled down to something simpler:

```
func FormatEventUserActor(u user.Info) string {
	if u.IsAdmin {
		return kargoapi.EventActorAdmin
	}
	if u.Username != "" {
		return kargoapi.EventActorOidcUsernamePrefix + u.Username
	}
	if emailClaim, ok := u.Claims["email"]; ok {
		if email, ok := emailClaim.(string); ok {
			return kargoapi.EventActorEmailPrefix + email
		}
	}
	if subClaim, ok := u.Claims["sub"]; ok {
		if sub, ok := subClaim.(string); ok {
			return kargoapi.EventActorSubjectPrefix + sub
		}
	}
	return kargoapi.EventActorUnknown
}
```

(fwiw, it was a tad more complex than necessary before you touched it.)